app/scripts/services/membership/membership.js

```diff
@@ -84,12 +84,12 @@ angular
       };
     };
 
+
     var mapRolebindingsForUI =  function(rolebindings, roles) {
       var mapForUI = _.reduce(
                       rolebindings,
                       function(result, rolebinding) {
-                        // hmm. dont like this here
-                        var roleKey = uniqueKey(rolebinding.roleRef.namespace ? 'Role' : 'ClusterRole', rolebinding.roleRef.name);
+                        var roleKey = uniqueKey(rolebinding.roleRef.kind, rolebinding.roleRef.name);
```

Comment:
Can we ever have role references from other namespaces? If so, the key won't be unique.

Reply:
I don't know if you can reference a role for namespace `foo` from namespace `bar`.   I think you can't, if its global it has to be a `ClusterRole`.

@enj true?

Reply:
Correct, but I would use all the data:

```go
// RoleRef contains information that points to the role being used
type RoleRef struct {
	// APIGroup is the group for the resource being referenced
	APIGroup string `json:"apiGroup" protobuf:"bytes,1,opt,name=apiGroup"`
	// Kind is the type of resource being referenced
	Kind string `json:"kind" protobuf:"bytes,2,opt,name=kind"`
	// Name is the name of resource being referenced
	Name string `json:"name" protobuf:"bytes,3,opt,name=name"`
}
```

Granted I am not sure what your code is actually doing...

Reply:
We might put it in anyway if it ever might be possible to make sure the key is always unique.

Reply:
OK so no namespace (which I guess is why we had the bug). Agree with @enj, though, I'd add the group.

Reply:
This is really a random property... I could probably make it a `Date.now()` even & skip all the actual values.  I think I used bits of the Role to keep it human readable for debugging.  Let me check & ensure & I can add anything to make it certainly unique.

Reply:
added `rolebinding.metadata.namespace` which will generate keys something like ' clusterrole-edit-myproject'.

Reply:
But that is always the same value for all bindings, so it doesn't help with uniqueness

Reply:
`roleRef` doesn't necessarily have a `namespace` (usually doesn't).  I can swap it for just a random number then.

Reply:
We're overthinking this... Just use the fields that are there: group + kind + name

Reply:
I believe `apiGroup` will always be the same, `rbac.authorization.k8s.io`, so I think we are probably coming back full circle to more or less what we have just with longer keys.

Reply:
That's true right now, but we can't assume. There's a property there that could be set to something else.